### PR TITLE
Grues

### PR DESCRIPTION
**Pull request by Shadowmech88**
Grues 2: Electric Boogaloo
Previous PR: #12520 

Grues deal 15 brute damage per hit, can't be stunned, and move at twice the speed of humans, but they take double damage from both brute and burn.
Grues also have inverted vision, meaning they see light as dark, and dark as light. This means that they can see perfectly in a pitch-black room, but can't see anything in a well-lit room.
image image

Grues have a spell they can use to create a negative light source centered on themselves, swallowing all light around them for up to a minute.
They also have a spell they can use to shatter all lights within seven tiles of them, with a two minute cooldown.
Grues can heal themselves by butchering humans with their claws and consuming human meat, 50 of each damage type is healed when they eat it.

I figure down the line they could make an interesting antag or random event or something.

This PR also adds a new charge type for spells that allows their charge to be drained gradually while the spell is active, and stop draining when the spell is deactivated, which happens when the spell runs out of charge, or can be done at any time by clicking the spell icon again.
It also makes humans butcherable. They still can't be gibbed via butchering, but you can get meat and teeth from them now. I'm not sure why that was disabled.
`/mob/living` now has a var that holds a list of tools that have been used to butcher it.
Autopsy scanners will now report what tools a human has been butchered with if the human's chest is scanned.
Being butchered by a grue, regardless of the tool used, will cause the autopsy scanner to report that the patient was eaten by a grue.

:cl:
 * rscadd: Added grues, only spawnable by admins.

### CODE REVIEW

**Comment by Shadowmech88** [comment]
New since last PR:
* Grues have a spell they can use to shatter all lights within seven tiles of them, with a two minute cooldown.
* `/mob/living` now has a var that holds a list of tools that have been used to butcher it.
* Autopsy scanners will now report what tools a human has been butchered with if the human's chest is scanned.
* Being butchered by a grue, regardless of the tool used, will cause the autopsy scanner to report that the patient was eaten by a grue.

**Comment by Shadowmech88** [comment]
Also I realized that an airlock-opening verb/spell wasn't necessary because grues are permahulked, so they can just punch a hole in the wall to get in and out of maintenance.

**Comment by Gunosaguki** [comment]
Hurray for grues

File: code/modules/spells/spell_code.dm [hunk 1]
```diff
@@ -84,6 +85,7 @@ var/list/spells = typesof(/spell) //needed for the badmin verb for now
 
 	var/obj/screen/spell/connected_button
 	var/currently_channeled = 0
+	var/gradual_casting = 0 //equals 1 while a Sp_GRADUAL spell is actively being cast
```
Comment (ComicIronic) [inline code/modules/spells/spell_code.dm]: Use TRUE and FALSE if you're introducing a new bool var.

File: code/modules/mob/living/living_defines.dm [hunk 2]
```diff
@@ -71,6 +71,8 @@
 	var/obj/screen/schematics_background
 	var/shown_schematics_background = 0
 
+	var/list/advanced_butchery = list() //Includes a list of the tools used to butcher the mob, detectable via autopsy scanner rather than examine.
```
Comment (ComicIronic) [inline code/modules/mob/living/living_defines.dm]: We really don't need a new list on every mob - I suggest having this start at null, and wrapping getting in a proc to only make the list as needed.

File: code/modules/spells/spell_code.dm (hunk 1 above)
Comment (ghost) [reply code/modules/spells/spell_code.dm]: It's not a standard until you standardise it 
There's absolutely no need to do this

Comment (ComicIronic) [reply code/modules/spells/spell_code.dm]: It makes the code easier to read. I'm not going to have an xth conversation about the use and utility of TRUE and FALSE over 1 and 0 in code. If you keep it up, I'm enforcing the types which do nothing on you..

Comment (ghost) [reply code/modules/spells/spell_code.dm]: It's not a standard until you standardise it.

**Comment by Shadowmech88** [comment]
This is ready to go now.

**Comment by Gunosaguki** [comment]
launch it! we need christmas grues named krampus

**Comment by Exxion** [comment]
We... already have Krampus...

**Comment by BobdaBiscuit** [comment]
You dun it now arry pottah.

**Pull request merged**